scanpy/readwrite.py

```diff
@@ -277,15 +277,53 @@ def _read_v3_10x_h5(filename, *, start=None):
                 (data, dsets['indices'], dsets['indptr']),
                 shape=(N, M),
             )
+            obs_dict = dict(obs_names=dsets['barcodes'].astype(str))
+            var_dict = dict(
+                var_names=dsets['name'].astype(str),
+            )
+
+            # Handles reading feature-barcode matrix
+            if 'gene_id' not in dsets:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['id'].astype(str),
+                    }
+                )
+            # Handles reading probe-barcode matrix
+            else:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['gene_id'].astype(str),
+                        'probe_ids': dsets['id'].astype(str),
+                    }
+                )
+            var_dict.update({'feature_types': dsets['feature_type'].astype(str)})
```

Comment:
Move this line inside `var_dict = dict(` at line 281.

Reply:
That fails a test. It needs `gene_id` to be a column before `feature_types`.

Reply:
Ah, got it. I wonder why…

Reply:
```suggestion
            var_dict['feature_types'] = dsets['feature_type'].astype(str)
```

Reply:
They must be using `.iloc[]` based access to get the `gene_ids` somewhere downstream.